etl/grapher_import.py

```diff
@@ -234,6 +234,23 @@ def upsert_table(
         return VariableUpsertResult(db_variable_id, source_id)
 
 
+def fetch_db_checksum(dataset: catalog.Dataset) -> Optional[str]:
+    """
+    Fetch the latest source checksum associated with a given dataset in the db. Can be compared
+    with the current source checksum to determine whether the db is up-to-date.
+    """
+    with open_db() as db:
```

Comment:
Fair point, yeah I think it's OK at current scale, especially since the connections are not concurrent. Every method in `grapher_import` already opens its own connection so I wanted to follow the pattern. Can think about reducing the number of connections while refactoring for ORM